llvm/lib/Transforms/Utils/InjectTLIMappings.cpp

```diff
@@ -35,35 +35,36 @@ STATISTIC(NumCompUsedAdded,
 
 /// A helper function that adds the vector function declaration that
 /// vectorizes the CallInst CI with a vectorization factor of VF
-/// lanes. The TLI assumes that all parameters and the return type of
-/// CI (other than void) need to be widened to a VectorType of VF
-/// lanes.
+/// lanes. For each mapping TLI contains a VABI prefix, which contains
+/// all information required to create vector function declaration.
 static void addVariantDeclaration(CallInst &CI, const ElementCount &VF,
-                                  bool Predicate, const StringRef VFName) {
+                                  const VecDesc *VD) {
   Module *M = CI.getModule();
+  FunctionType *FTy = CI.getFunctionType();
```

Comment:
nit: I would rename this to `ScalarFTy` to match similar code that uses VFABI.

Reply:
Done